src/modules/campaign-monitor/services/campaign-monitor.service.ts

```diff
@@ -95,17 +100,78 @@ export class CampaignMonitorService {
 		return true;
 	}
 
-	public async sendMail(
-		emailInfo: CampaignMonitorSendMailDto
-	): Promise<void | BadRequestException> {
-		let url: string | null = null;
+	// TODO: Write tests (ARC-1500)
+	public async sendForMaterialRequest(emailInfo: MaterialRequestEmailInfo): Promise<boolean> {
+		const recipients: string[] = [];
+
+		if (emailInfo.to) {
+			recipients.push(emailInfo.to);
+		} else {
+			this.logger.error(
+				`Mail will not be sent to maintainer id ${emailInfo.materialRequests[0].maintainerId} - empty email address`
+			);
+		}
+
+		if (recipients.length === 0) {
+			this.logger.error(
+				`Mail will not be sent - no recipients. emailInfo: ${JSON.stringify(emailInfo)}`
+			);
+			return false;
+		}
+
+		const cmTemplateId = templateIds[emailInfo.template];
+		if (!cmTemplateId) {
+			this.logger.error(
+				`Campaign monitor template ID for ${emailInfo.template} not found -- email could not be sent`
+			);
+			return false;
+		}
+
+		const url = `/${
+			process.env.CAMPAIGN_MONITOR_TRANSACTIONAL_SEND_MAIL_API_VERSION as string
+		}/${process.env.CAMPAIGN_MONITOR_TRANSACTIONAL_SEND_MAIL_API_ENDPOINT as string}${
+			templateIds[emailInfo.template]
+		}/send`;
+
+		const data: CampaignMonitorData = {
+			to: recipients,
+			consentToTrack: 'unchanged',
```

Comment:
Why is this a separate send function for material requests?

Shouldn't all this code be checked for any email we want to send? I would expect this to be in the sendTransactionalMail function.

We can keep the sendForMaterialRequest function, but only for mapping info to the email template

Reply:
The way how the recipients are checked(line 107-113) is different than how it is in sendVisit, so this should be in the sendForMaterialRequest function. I do think that everything from line 115 to 128 could be in the sendTransactionalMail function.